.github/workflows/ci-test.yml

```diff
@@ -21,6 +21,9 @@ jobs:
     steps:
       - uses: actions/checkout@v2
         with:
+          # This ensures that the tip of the PR is checked out instead of the merge between the base ref and the tip
+          # On `push` this value will be empty and will "do-the-right-thing"
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Huh, I don't know what this means at all... what does "checkout tip of PR instead of merge commit" mean? What's the "merge commit" in this context? I'm quite shocked checking out the head of the PR isn't the default behavior of `actions/checkout`.

https://github.com/actions/checkout#usage says:

```
    # The branch, tag or SHA to checkout. When checking out the repository that
    # triggered a workflow, this defaults to the reference or SHA for that event.
    # Otherwise, uses the default branch.
```

This seems like the same thing as the head commit? What's the difference here? Could you help by drawing a DAG?

Reply:
This is because gha checkout be default doesn't pull a sha from the repo is makes it's own squash - and has no history at all in the pull.